### PR TITLE
RFC: using window.matchMedia

### PR DESCRIPTION
**Pull request by necolas**
I had to work around not being able to transfer data from the listener to the getter, and neither accepting custom arguments. Let me know if you have any better ideas.

### CODE REVIEW

File: examples/media-query/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+const mediaQueries = {
+  small: "(min-width:300px)",
+  medium: "(min-width: 400px)",
+  large: "(min-width: 500px)"
+}
+
+render(
+  <MediaProvider
+    getMedia={() => mediaQueryGetter(mediaQueries)}
```
Comment (bloodyowl) [inline examples/media-query/index.js:30]: let's just use a closure there, `getMedia={createMediaQueryGetter(mediaQueries)}`

File: examples/media-query/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    listener={(update) => mediaQueryListener(mediaQueries, update)}>
```
Comment (bloodyowl) [inline examples/media-query/index.js:31]: here too, `listener={createMediaQueryListener(mediaQueries)}`

File: src/mediaQueryListener.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const mediaQueryListener = (mediaQueries, update) => {
+  const mqlList = Object.keys(mediaQueries).map(size => {
+    const mql = window.matchMedia(mediaQueries[size])
+    mql.addListener(update)
+    return mql
+  })
+
+  return () => {
```
Comment (bloodyowl) [inline src/mediaQueryListener.js:8]: in terms of coding style I'd prefer `() => mqlList.forEach((mql) => mql.removeListener(update))`

**Comment by bloodyowl** [comment]
this looks really nice, just a couple of inline comments

**Comment by MoOx** [comment]
Sweet indeed.

**Pull request merged**